angel.py

```diff
@@ -75,131 +78,186 @@ def read_csv(filename):
     return person_list
 
 
-def convert_to_player(row):
+def generalized_convert_to_player(row, column_names_index_dict):
+    player_data = []
+
     try:
-        config_file = open("config.yml", 'r')
-        index_dict = yaml.full_load(config_file)["player_attribute_index"]
-
-        player_username = row[index_dict["telegram_username"]].strip().lower()
-        player_name = row[index_dict["name"]].strip().lower()
-        gender_pref = row[index_dict["gender_pref"]].strip().lower()
-        gender_player = row[index_dict["gender"]].strip().lower()
-        interests = row[index_dict["interests"]].strip()
-        two_truths_one_lie = row[index_dict["two_truths_one_lie"]].strip()
-        introduction = row[index_dict["introduction"]].strip()
-        house_number = row[index_dict["house_number"]].strip().lower()
-        cg_number = row[index_dict["cg_number"]].strip().lower()
-        year_of_study = row[index_dict["year_of_study"]].strip().lower()
-        faculty = row[index_dict["faculty"]].strip().lower()
+        special_columns = column_names_index_dict["special columns"]
+        enum_columns = column_names_index_dict["enum columns"]
+        text_columns = column_names_index_dict["text columns"]
+
+        player_data = (generate_column_key_value_pair(special_columns, row) +
+                       generate_column_key_value_pair(text_columns, row) +
+                       generate_column_key_value_pair(enum_columns, row, check_enum_columns_input_valid))
```

Comment:
I don't see why you would handle the different kinds of columns using the same function. Why not have three separate functions?
- handle special
- handle text
- handle enum